collector/workitem.go

```diff
@@ -0,0 +1,337 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/fabric8-services/fabric8-notification/wit"
+	"github.com/fabric8-services/fabric8-notification/wit/api"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/goadesign/goa/uuid"
+)
+
+func NewCommentResolver(c *api.Client) ReceiverResolver {
+	return func(ctx context.Context, id string) ([]Receiver, map[string]interface{}, error) {
+		cID, err := uuid.FromString(id)
+		if err != nil {
+			return []Receiver{}, nil, fmt.Errorf("Could not lookup comment based on id %v", id)
+		}
+		return Comment(ctx, c, cID)
+	}
+}
+
+func NewWorkItemResolver(c *api.Client) ReceiverResolver {
+	return func(ctx context.Context, id string) ([]Receiver, map[string]interface{}, error) {
+		wID, err := uuid.FromString(id)
+		if err != nil {
+			return []Receiver{}, nil, fmt.Errorf("Could not lookup Workitem based on id %v", id)
+		}
+		return WorkItem(ctx, c, wID)
+	}
+}
+
+func Comment(ctx context.Context, c *api.Client, cID uuid.UUID) ([]Receiver, map[string]interface{}, error) {
+	var values = map[string]interface{}{}
+	var errors []error
+	var users []uuid.UUID
+
+	comment, err := wit.GetComment(ctx, c, cID)
+	if err != nil {
+		return []Receiver{}, nil, err
+	}
+	values["comment"] = comment
+	users = append(users, collectCommentUsers(comment)...)
+
+	commentOwner, err := wit.GetUser(ctx, c, *comment.Data.Relationships.CreatedBy.Data.ID)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	values["commentOwner"] = commentOwner
+
+	wiID, _ := uuid.FromString(*comment.Data.Relationships.Parent.Data.ID)
+
+	wi, err := wit.GetWorkItem(ctx, c, wiID)
+	if err != nil {
+		return []Receiver{}, nil, err
+	}
+	users = append(users, collectWorkItemUsers(wi)...)
+	values["workitem"] = wi
+
+	ownerID, _ := uuid.FromString(*wi.Data.Relationships.Creator.Data.ID)
```

Comment:
You do it on purpose, but we might need to check the errors in this conversion :)